go/vt/vtgate/planbuilder/querytree_transformers.go

```diff
@@ -430,10 +428,18 @@ func transformRoutePlan(ctx *planningContext, n *routeTree) (*route, error) {
 	}
 
 	var singleColumn vindexes.SingleColumn
-	var values []sqltypes.PlanValue
+	var value engine.RouteValue
 	if n.selectedVindex() != nil {
 		singleColumn = n.selected.foundVindex.(vindexes.SingleColumn)
-		values = n.selected.values
+		if len(n.selected.values) == 1 {
```

Comment:
is this safe to do? @harshit-gangal

Reply:
the length is always going to be `1` for singleColumn Vindex. This check is not required.